mop/brokers/asassn.py

```diff
@@ -0,0 +1,224 @@
+
+    def open_webpage(self):
+        page_status_code=requests.get(BROKER_URL).status_code
+        return page_status_code
+
+    def retrieve_transient_table(self):
+        page=requests.get(BROKER_URL)
+        doc = lh.fromstring(page.content)
+        tr_elements = doc.xpath('//tr')
+        col=[]
+        i=0
+        for t in tr_elements[0]:
+            i+=1
+            content=t.text_content()
+            col.append((content,[]))
+        for j in range(1,len(tr_elements)):
+            #T is our j'th row
+            T=tr_elements[j]
+            #If row is not of size 12, the //tr data is not from our table 
+            if len(T)!=12:
+                break
+            i=0
+            #Iterate through each element of the row
+            for t in T.iterchildren():
+                data=t.text_content() 
+                #Check if row is empty
+                if i>0:
+                #Convert any numerical value to integers
+                    try:
+                        data=int(data)
+                    except:
+                        pass
+                #Append the data to the empty list of the i'th column
+                col[i][1].append(data)
+                #Increment i for the next column
+                i+=1
+        return col
+
+    def retrieve_microlensing_coordinates(self):
+        transienttable = self.retrieve_transient_table()
+        listofindices=[]
+        i=0
+        newlist = [str(s) for s in transienttable[11][1]]   
+        for m in newlist:
+            if 'microlensing' in m or 'Microlensing' in m:
+                listofindices.append(i)
+            i+=1
+        fullralist = []
+        fulldeclist = []
+        fullids = []
+        fullasassnids = []
+        fullralist = [str(s) for s in transienttable[3][1]] 
+        fulldeclist = [str(s) for s in transienttable[4][1]]
+        fullids = [str(s) for s in transienttable[0][1]]
+        fullasassnids = [str(s) for s in transienttable[1][1]]
+        selectedralist=[]
+        selecteddeclist=[]
+        selectedids=[]
+        selectedasassnids=[]
+        listofevents=[]
+        for n in listofindices:
+            #is this set up efficiently? 
+            listofevents.append([fullids[n],fullasassnids[n],fullralist[n],fulldeclist[n]])
+        return listofevents
+    
+    def fetch_alerts(self):
+        list_of_targets=[]
+        time_now=Time(datetime.datetime.now()).jd
+        listofevents = self.retrieve_microlensing_coordinates()
+        #first index for listofevents is for first event
+        #subindices are for id, second id, ra, dec
+        #event is an array
+        for event in listofevents[0:]:
+            target_name=event[0]
+            ra_split=event[2].split(':')
+            dec_split=event[3].split(':')
+            ra = float(ra_split[0]) + (float(ra_split[1])/60) + (float(ra_split[2])/3600)
+            dec = float(dec_split[0]) + (float(dec_split[1])/60) + (float(dec_split[2])/3600)
```

Comment:
Beware this may not handle negative declinations correctly!  SkyCoord objects can be generated directly from sexigesimal-format strings and has been verified to properly deal with this issue, so best to use that approach rather than pre-convert to floats.